### PR TITLE
test(scale): [NPM] fix flakes in kwok and capture kernel state on failure

### PR DESCRIPTION
**Pull request by huntergregory**
**Reason for Change**:
KWOK Pods were not being scheduled when running KWOK in a background process. Now, we run KWOK as a Pod.

Also, change logic to wait for pods to run before creating NetPols.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] relevant PR labels added

**Notes**:

### CODE REVIEW

**Comment by huntergregory** [comment]
Windows Scale Test typically fails now because HNS latencies have seemed to increase

**Review by rayaisaiah — APPROVED** [review]

File: test/scale/test-scale.sh [hunk 1]
```diff
@@ -489,16 +525,18 @@ if [[ $numUniqueLabelsPerPod -gt 0 ]]; then
     done
 fi
 
-set -x
+wait_for_pods
```
Comment (huntergregory) [inline test/scale/test-scale.sh:528]: to better evaluate time to apply ACLs, wait for pods to come up first (takes a variable amount of time) before applying the NetPols

File: test/scale/test-scale.sh [hunk 2]
```diff
@@ -276,21 +276,47 @@ fi
...
+            echo "waiting for fake pods to run (try $count)"
+            count=$((count+1))
+            # just make sure kwok pods are Running, not necessarily Ready (sometimes kwok pods have NodeNotReady even though the node is ready)
+            set +e -x
+            $KUBECTL $KUBECONFIG_ARG wait --for=condition=Initialized pods -n scale-test -l is-kwok=true --all --timeout=0 && set -e +x && break
+            endDate=`date +%s`
+            set -e +x
+            if [[ $endDate -gt $(( startDate + (20*60) )) ]]; then
+                echo "timed out waiting for all kwok pods to run"
+                k get pod -n scale-test -owide
```
Comment (huntergregory) [inline test/scale/test-scale.sh:293]: k top pod/node

**Review by vakalapa — APPROVED** [review]

**Pull request merged**